databricks/koalas/base.py

```diff
@@ -171,6 +173,11 @@ def spark_column(self):
     __neg__ = column_op(Column.__neg__)
 
     def __add__(self, other):
+        if isinstance(self.spark.data_type, NumericType):
```

Comment:
`not isinstance(self.spark.data_type, StringType)` might be better?

Reply:
Makes sense, updated!